File: Nx3Java/java/src/nx3/PNoteHeadsRectTplCalculator.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PNoteHeadsRectTplCalculator extends haxe.lang.HxObject
{
	public    PNoteHeadsRectTplCalculator(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PNoteHeadsRectTplCalculator(nx3.PNote note)
	{
		nx3.PNoteHeadsRectTplCalculator.__hx_ctor_nx3_PNoteHeadsRectTplCalculator(this, note);
	}
	
	
	public static   void __hx_ctor_nx3_PNoteHeadsRectTplCalculator(nx3.PNoteHeadsRectTplCalculator __temp_me121, nx3.PNote note)
	{
		__temp_me121.note = note;
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.PNoteHeadsRectTplCalculator(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.PNoteHeadsRectTplCalculator(((nx3.PNote) (arr.__get(0)) ));
	}
	
	
	public  nx3.PNote note;
	
	public   haxe.root.Array<nx3.geom.Rectangle> getHeadsRects()
	{
		return new haxe.root.Array<nx3.geom.Rectangle>(new nx3.geom.Rectangle[]{new nx3.geom.Rectangle(((java.lang.Object) (-6) ), ((java.lang.Object) (-5.3) ), ((java.lang.Object) (10) ), ((java.lang.Object) (8.8) ))});
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef717 = true;
			switch (field.hashCode())
			{
				case 3387378:
				{
					if (field.equals("note")) 
					{
						__temp_executeDef717 = false;
						this.note = ((nx3.PNote) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef717) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef718 = true;
			switch (field.hashCode())
			{
				case 2051530802:
				{
					if (field.equals("getHeadsRects")) 
					{
						__temp_executeDef718 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getHeadsRects"))) );
					}
					
					break;
				}
				
				
				case 3387378:
				{
					if (field.equals("note")) 
					{
						__temp_executeDef718 = false;
						return this.note;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef718) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef719 = true;
			switch (field.hashCode())
			{
				case 2051530802:
				{
					if (field.equals("getHeadsRects")) 
					{
						__temp_executeDef719 = false;
						return this.getHeadsRects();
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef719) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("note");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}

File: Nx3CS/bin/test/src/nx3/PNoteHeadsRectTplCalculator.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class PNoteHeadsRectTplCalculator : global::haxe.lang.HxObject {
		public    PNoteHeadsRectTplCalculator(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PNoteHeadsRectTplCalculator(global::nx3.PNote note){
			unchecked {
				#line 13 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
				global::nx3.PNoteHeadsRectTplCalculator.__hx_ctor_nx3_PNoteHeadsRectTplCalculator(this, note);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_PNoteHeadsRectTplCalculator(global::nx3.PNoteHeadsRectTplCalculator __temp_me106, global::nx3.PNote note){
			unchecked {
				#line 15 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
				__temp_me106.note = note;
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
				return new global::nx3.PNoteHeadsRectTplCalculator(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
				return new global::nx3.PNoteHeadsRectTplCalculator(((global::nx3.PNote) (arr[0]) ));
			}
			#line default
		}
		
		
		public  global::nx3.PNote note;
		
		public virtual   global::Array<object> getHeadsRects(){
			unchecked {
				#line 20 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
				return new global::Array<object>(new object[]{new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (-6) ), true), new global::haxe.lang.Null<double>(-5.3, true), new global::haxe.lang.Null<double>(((double) (10) ), true), new global::haxe.lang.Null<double>(8.8, true))});
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
				switch (hash){
					case 1225398258:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
						this.note = ((global::nx3.PNote) (@value) );
						#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
						return @value;
					}
					
					
					default:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
				switch (hash){
					case 325402290:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getHeadsRects") ), ((int) (325402290) ))) );
					}
					
					
					case 1225398258:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
						return this.note;
					}
					
					
					default:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
				switch (hash){
					case 325402290:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
						return this.getHeadsRects();
					}
					
					
					default:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
				baseArr.push("note");
				#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
				{
					#line 9 "F:\\nx3\\src\\nx3\\PNoteHeadsRectTplCalculator.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}


